File: Lista-15/java/6.java
class Motor {
    private int potencia;

    public Motor(int potencia) {
        this.potencia = potencia;
    }

    public int getPotencia() {
        return this.potencia;
    }
}

class Carro {
    private String marca;
    private String modelo;
    private Motor motor;

    public Carro(String marca, String modelo, Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.motor = motor;
    }

    public String detalhes() {
        return "Carro: " + this.marca + ", " + this.modelo + ", Motor: " + this.motor.getPotencia() + " HP";
    }

    public static void main(String[] args) {
        Motor motor = new Motor(150);
        Carro carro = new Carro("Toyota", "Corolla", motor);
        System.out.println(carro.detalhes());
    }
}

File: Lista-15/go/6.go
package main
import "fmt"

type Motor struct {
    Potencia int
}

type Carro struct {
    Marca   string
    Modelo  string
    Motor   Motor
}

func (c Carro) Detalhes() string {
    return fmt.Sprintf("Carro: %s, %s, Motor: %d HP", c.Marca, c.Modelo, c.Motor.Potencia)
}

func main() {
    motor := Motor{150}
    carro := Carro{"Toyota", "Corolla", motor}
    fmt.Println(carro.Detalhes())
}
